data_safe_haven/infrastructure/project_manager.py

```diff
@@ -170,19 +167,18 @@ def cancel(self) -> None:
                 f"Cancelling ongoing Pulumi operation for stack [green]{self.stack.name}[/]."
             )
             self.stack.cancel()
+            self.logger.warning(
+                f"Removing any ambiguous Pulumi resources from stack [green]{self.stack.name}[/]."
+            )
+            self.run_pulumi_command("refresh --clear-pending-creates --yes")
```

Comment:
We can't use `self.stack.refresh()`?

Reply:
1. I don't think we want the default behaviour of refresh to include `--clear-pending-creates` 
2. There's no option to `automation.Stack.refresh` that adds this option so we'd have to write our own `refresh` function.